src/cli.py

```diff
@@ -1,19 +1,31 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
-
+import json
 import logging
 import re
+from dataclasses import asdict, dataclass, field
 from typing import List, Optional
 
 from ops.model import Container
 from ops.pebble import Error, ExecError
 
+from constants import WORKLOAD_SERVICE
+from env_vars import EnvVars
+
 VERSION_REGEX = re.compile(r"App Version:\s*(?P<version>\S+)\s*$")
 MODEL_REGEX = re.compile(r"Created model:\s*(?P<model>\S+)")
+IDENTITY_REGEX = re.compile(r"Identity created:\s*(?P<identity>\S+)")
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class CmdOptions:
+    environment: EnvVars = field(default_factory=dict)
+    timeout: int = 20
+    stdin: Optional[str] = None
```

Comment:
issue: two things here
- env technically is not part of the command options, so if this data class needs to represent that, environment should not be part of this
- `stdin` as a string is very misleading, since this represents the content of the identity specification in json format (as a string)

If you're trying to implement the command pattern then you should have one class for each function input, which at this stage would really be too much.
I suggest you either refactor this, and depending on the meaning you give to your dataclass, then environment variables should either be left in there or taken out.

Reply:
I am not planning to implement command pattern. Command pattern is definitely an overkill here. What I am trying to do is simply have a place to wrap the extra options for the command run. So that we don't have to add/remove arguments for the `_run_cmd`.

Reply:
Based on the limited number of "options" we will use so far in all charms, I think it will be just fine. The dataclass will not explode in a foreseeable period.

I changed the type hint for the `stdin` to more represent what it is.